### ISSUE TITLE
Recommend setting default ordering when using UUID primary keys

### ISSUE THREAD

**Issue by matthewmcgarvey** [luckyframework/website#548]
PostgreSQL orders by primary key by default which makes complete sense when the primary key is an incrementing number. Calling `UserQuery.first` will always return the expected record in those cases. When the primary key is set to be a UUID, this becomes an issue. Without looking it up, I think it would then be ordered as you would sort strings so "A" comes before "B". Since UUIDs are random, the first UUID generated could start with any letter or number which means that the sorting could change over time. Calling `UserQuery.first` could return a different record tomorrow or the day after. 

We should add documentation to set a default ordering in queries using UUID primary keys.

It would look like:

```crystal
class UserQuery < User::BaseQuery
  def initialize
    defaults &.created_at.asc_order
  end
end
```

While Avram does provide a default ordering, it only does it if an ordering is not set
https://github.com/luckyframework/avram/blob/4f552db1c3769c741cffeb5233c91e93072b3c3e/src/avram/primary_key_queryable.cr#L22-L28

**Comment by jkthorne** [comment]
I think `created_at` is a better default then `id` personally.  I would just make `created_at` the default for `first`.

**Comment by matthewmcgarvey** [comment]
@wontruefree It might be but we don't have anything in place to know whether or not they have that column and I'd be worried about that being unexpected behavior

**Comment by stephendolan** [comment]
Yeah, oftentimes for basic join tables (Video <- VideoTopic -> Topic), I don't put `created_at` or `updated_at` on the middleman.

**Comment by robcole** [comment]
https://crystalshards.org/shards/github/wyhaines/csuuid.cr Is an interesting approach to UUIDs that might help reduce this as a concern as well.

**Comment by robcole** [comment]
> I think `created_at` is a better default then `id` personally. I would just make `created_at` the default for `first`.

Not sure if it is in Lucky, but time stamps usually aren’t indexed in other frameworks I’ve used, so you’d generally want to avoid ordering by those.
